apps/ex_wire/lib/ex_wire/kademlia/routing_table.ex

```diff
@@ -240,23 +289,39 @@ defmodule ExWire.Kademlia.RoutingTable do
     Enum.at(buckets, id)
   end
 
-  @spec traverse(t(), integer(), [Node.t()], integer()) :: [Node.t()]
-  defp traverse(table, bucket_id, acc \\ [], step \\ 1) do
+  @spec traverse(t(), Keyword.t()) :: [Node.t()]
+  defp traverse(table, opts) do
```

Comment:
hmmm this function is getting pretty complicated. Maybe pull out `found_nodes(bucket_id, step)` and `is_within_boundry(bucket_id, step)` into private methods?